<launch>
    <arg name="namespace"   default="red"/>
    <arg name="odom"        default="/mavros/global_position/local"/>
    <arg name="traj_ref"    default="uav/trajectory"/>

    <group ns="$(arg namespace)">
        
        <!-- Visual servo node -->
        <node
            pkg="uav_ros_control"
            type="visual_servo_node"
            name="visual_servo_node"
            output="screen">

            <!-- Topic remapping -->

            <remap from="x_error" to="red_color_filter/yaw_err"/>
            <remap from="y_error" to="red_color_filter/pitch_err"/>
            <remap from="odometry" to="$(arg odom)"/>
            <remap from="imu" to="/mavros/imu/data"/>
            <remap from="yaw_error" to="/red/red_color_filter/orientation_err"/>
            <remap from="n_contours" to="/red/red_color_filter/nContours"/>
            
        </node>
    </group>
</launch>
